File: reservation-frontend/src/app/request-commands/editEventOccurrenceCommand.ts
import { Reservation } from "../models/reservation"
import { Ticket } from "../models/ticket"

export class EditEventOccurrenceCommand {

    constructor(
        id: number, 
        startTime: Date, 
        eventId: number, 
        isActive: boolean, 
        isDeleted: boolean, 
        tickets: Ticket[], 
        reservations: Reservation[]) {
        this.id = id
        this.startTime = startTime
        this.eventId = eventId
        this.isActive = isActive
        this.isDeleted = isDeleted
        this.tickets = tickets
        this.reservations = reservations
    }

    public id!: number
    public startTime!: Date
    public eventId!: number
    public isActive!: boolean
    public isDeleted!: boolean
    public tickets!: Ticket[] 
    public reservations!: Reservation[]
}

File: reservations-api/src/Reservations.Application/EventOccurrences/Commands/EditEventOccurrence/EditEventOccurrenceCommand.cs
using MediatR;
using Reservations.Application.DataTransferObjects;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservations.Application.EventOccurrences.Commands.EditEventOccurrence
{
    public class EditEventOccurrenceCommand : IRequest
    {
        public long Id { get; set; }
        public DateTime StartTime { get; set; }
        public long EventId { get; set; }
        public bool IsActive { get; set; }
        public bool IsDeleted { get; set; }
        public IEnumerable<TicketDto> Tickets { get; set; }
        public IEnumerable<ReservationDto> Reservations { get; set; }
    }
}
